prometheus-query/src/v1/queries/instant.rs

```diff
@@ -0,0 +1,122 @@
+//! Impelement instant queries
+
+use super::{Client, QueryResult};
+use failure::Fallible;
+use reqwest;
+use std::time::Duration;
+
+pub static INSTANT_QUERY_PATH_SUFFIX: &str = "/api/v1/query";
+
+impl Client {
+    /// Sends the given query to the remote API, given an optional `time` and timeout.SystemTime
+    ///
+    /// The `time` is measured since the UNIX_EPOCH
+    pub async fn query(
+        &self,
+        query: String,
+        time: Option<chrono::DateTime<chrono::Utc>>,
+        timeout: Option<Duration>,
+    ) -> Fallible<QueryResult> {
+        self.new_request(reqwest::Method::GET, INSTANT_QUERY_PATH_SUFFIX)
+            .and_then(move |request_builder| {
+                let mut query = vec![("query", query)];
+
+                if let Some(time) = time {
+                    query.push(("time", time.to_rfc3339()));
+                }
+
+                if let Some(timeout) = timeout {
+                    query.push(("timeout", format!("{}s", timeout.as_secs())));
+                };
+
+                trace!("sending query '{:?}'", &query);
+                request_builder.query(&query).send().map_err(Into::into)
+            })
+            .and_then(|response| response.error_for_status().map_err(Into::into))
+            .and_then(|mut response| response.json().map_err(Into::into))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[cfg(feature = "test-net-private")]
+    #[ignore]
+    #[test]
+    fn query_infogw() -> Fallible<()> {
```

Comment:
Are you interested in fixing and activating this, or some other, test?

Reply:
Oh, right, certainly - added a TODO

Reply:
A TODO on the PR level means that fixing and activating these tests are in scope of this PR, right?

Reply:
So tests compiles, but it requires PROMETHEUS_TOKEN to function. 

Not sure how to proceed - I guess it should be moved to e2e suite and use in-cluster prometheus instead of production URL?

Reply:
> I guess it should be moved to e2e suite and use in-cluster prometheus instead of production URL?

What controls the life-cycle of the in-cluster prometheus? Will it only include data from the current test-run -- which would be ideal? If so, that sounds like an ideal solution to me.

Reply:
>What controls the life-cycle of the in-cluster prometheus?

It would live as long as the cluster itself

>Will it only include data from the current test-run

Yep

I guess the safest option would be:
a) move to e2e
b) re-use `PROM_ENDPOINT`/`PROM_TOKEN` env vars for this test
c) query something simple like `up{job="apiserver"}` there

Reply:
This would require Prometheus to return expected timestamps and labels (like node names), which are being generated every e2e run. So the test currently compiles, but I'm not running it in e2e suite

Reply:
Could we come up with a query which has an at least partially static result, so we can only verify that part?

Reply:
We'd still have to rip out timestamps, so it would take more work to enable it